ensembl_genes/Red.py

```diff
@@ -0,0 +1,264 @@
+                                                         'logic_name':self.param('logic_name'), \
+                                                         'program':'Red', \
+                                                         'program_version':'05/22/2015', \
+                                                         'program_file':self.param('red_path')})
+        connection.execute(analysis_insert)
+
+        # fetch the inserted analysis_id
+        analysis_query = db.select([analysis_table.columns.analysis_id])
+        analysis_query = analysis_query.where(analysis_table.columns.logic_name == 'red')
+        analysis_results = connection.execute(analysis_query).fetchall()
+        analysis_id = analysis_results[0][0]
+
+        # insert repeat analysis meta keys
+        meta_insert = meta_table.insert(None).values({'species_id':1, \
+                                                  'meta_key':'repeat.analysis', \
+                                                  'meta_value':'red'})
+        connection.execute(meta_insert)
+
+        # insert dummy repeat consensus
+        repeat_consensus_insert = repeat_consensus_table.insert(None).values({'repeat_name':'Red', \
+                                                                          'repeat_class':'Red', \
+                                                                          'repeat_type':'Red', \
+                                                                          'repeat_consensus':'N'})
+        connection.execute(repeat_consensus_insert)
+
+        # fetch the inserted repeat_consensus_id
+        repeat_consensus_query = db.select([repeat_consensus_table.columns.repeat_consensus_id])
+        repeat_consensus_query = repeat_consensus_query.where(repeat_consensus_table.columns.repeat_name == 'Red')
+        repeat_consensus_results = connection.execute(repeat_consensus_query).fetchall()
+        repeat_consensus_id = repeat_consensus_results[0][0]
+
+        # parse the repeats file and make a tsv file ready to load into the repeat_feature table
+        repeats_file = self.parse_repeats(self.param('rpt'),repeat_consensus_id,analysis_id)
+
+        # insert repeat features
+        repeat_feature_query = "LOAD DATA LOCAL INFILE '"+repeats_file+"' \
+                                INTO TABLE repeat_feature \
+                                FIELDS TERMINATED BY '\\t' \
+                                LINES TERMINATED BY '\\n' \
+                                (seq_region_id,seq_region_start,seq_region_end,\
+                                 repeat_start,repeat_end,repeat_consensus_id,analysis_id)"
+        connection.execute(repeat_feature_query)
+
+
+    def parse_repeats(self,rpt,repeat_consensus_id,analysis_id):
+        """ It parses the Red's program output and it converts it into
+            a tsv file which can be loaded into an Ensembl core repeat_feature table. """
```

Comment:
This one works for any methods. The python group is still unsure of how to write documentation. Compara mentionned Google style https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html. Some thinks it is too much

Reply:
Yes, I think we should discuss and choose a standard template for the docstrings for methods. I don't have any special preference atm.